clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -931,7 +929,23 @@ public void testMissingOffsetNoResetPolicy(GroupProtocol groupProtocol) {
 
         // lookup committed offset and find nothing
         client.prepareResponseFrom(offsetResponse(Collections.singletonMap(tp0, -1L), Errors.NONE), coordinator);
-        assertThrows(NoOffsetForPartitionException.class, () -> consumer.poll(Duration.ZERO));
+
+        if (groupProtocol == GroupProtocol.CONSUMER) {
+            // New consumer poll(ZERO) needs to wait for the offset fetch event add by a call to poll, to be processed
+            // by the background thread, so it can realize there are no committed offsets and then
+            // throw the NoOffsetForPartitionException
+            TestUtils.waitForCondition(() -> {
+                try {
+                    consumer.poll(Duration.ZERO);
+                    return false;
+                } catch (NoOffsetForPartitionException e) {
+                    return true;
+                }
+            }, "Consumer was not able to update fetch positions on continuous calls with 0 timeout");
+        } else {
+            // CLASSIC CONSUMER PROTOCOL
```

Comment:
nit: I take this as a bit confusing, given that it's truly the `LegacyConsumer` the one that causes the difference in this test (not the protocol). I would personally remove it, don't see much value in it. (not hinting to use the LegacyConsumer in the comment either, that won't age well, it will be renamed soon https://issues.apache.org/jira/browse/KAFKA-17060)

Reply:
Thanks information. Updated.